synapse/handlers/federation_event.py

```diff
@@ -1036,9 +1056,16 @@ async def _get_state_ids_after_missing_prev_event(
         # XXX: this doesn't sound right? it means that we'll end up with incomplete
         #   state.
         failed_to_fetch = desired_events - event_metadata.keys()
+
+        # The event_id is part of the `desired_events` but isn't fetched as part
+        # of the `event_metadata` so we remove it here separately if we did find it.
+        have_event_id = await self._store.have_seen_event(room_id, event_id)
+        if have_event_id:
+            failed_to_fetch = failed_to_fetch - {event_id}
```

Comment:
Do we even care that we have `event_id` locally at this point in time? Should we just always subtract `{event_id}` from `failed_to_fetch`?

Reply:
Looks like https://github.com/matrix-org/synapse/pull/13258 already decided on dropping it as I suspected here 😁